operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/workflow/CRDPresentActivationCondition.java

```diff
@@ -0,0 +1,108 @@
+package io.javaoperatorsdk.operator.processing.dependent.workflow;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.apiextensions.v1.CustomResourceDefinition;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+import io.javaoperatorsdk.operator.api.reconciler.dependent.DependentResource;
+import io.javaoperatorsdk.operator.processing.GroupVersionKind;
+import io.javaoperatorsdk.operator.processing.expiration.Expiration;
+import io.javaoperatorsdk.operator.processing.expiration.ExpirationExecution;
+import io.javaoperatorsdk.operator.processing.expiration.RetryExpiration;
+import io.javaoperatorsdk.operator.processing.retry.GenericRetry;
+import io.javaoperatorsdk.operator.processing.retry.Retry;
+
+public class CRDPresentActivationCondition implements Condition<HasMetadata, HasMetadata> {
```

Comment:
I think this should be better implemented as a utilty class that could be reused  elsewhere and shared across reconcilers, in particular because Controller also can check the availability of the CRD so it would be better if the same logic / implementation / cache could be reused/shared everywhere this functionality is needed.

Reply:
Yes, that is true, I don't like utilities / static scoped states, usually, those lead to problems (eventually). But will think about how to handle that.